### PR TITLE
libobs: Add Windows X64 Emulation on ARM64 and move Rosetta Detection

### PR DESCRIPTION
**Pull request by tommyvct**
### Description

Adds a check that detects if OBS is running on Windows ARM64 emulation and logs the result.

### Motivation and Context

https://github.com/obsproject/obs-studio/pull/5644
I messed up #5738, this is the new PR. 

### How Has This Been Tested?

Running X64 OBS Studio gives the following log:

#### Windows:

<details>

```
23:10:09.304: CPU Name: Microsoft SQ1 @ 3.0 GHz
23:10:09.305: CPU Speed: 1670MHz
23:10:09.305: Physical Cores: 8, Logical Cores: 8
23:10:09.305: Physical Memory: 7718MB Total, 3202MB Free
23:10:09.305: Windows Version: 10.0 Build 22538 (release: 2009; revision: 1010; ARM 64-bit)
23:10:09.305: Windows ARM64: Running with X64 Emulation
23:10:09.306: Running as administrator: false
23:10:09.307: Aero is Enabled (Aero is always on for windows 8 and above)
23:10:09.307: Windows 10 Gaming Features:
23:10:09.307: 	Game Bar: On
23:10:09.307: 	Game DVR: On
23:10:09.307: 	Game DVR Background Recording: Off
23:10:09.317: Current Date/Time: 2022-01-25, 23:10:09
23:10:09.317: Browser Hardware Acceleration: false
23:10:09.317: Qt Version: 5.15.2 (runtime), 5.15.2 (compiled)
23:10:09.317: Portable mode: false
23:10:10.338: OBS 27.2.0-beta4-23-g026987a2b (64-bit, windows)
```

</details>

#### macOS:

<details>

```
17:58:05.833: CPU Name: Apple M1 Pro
17:58:05.833: CPU Speed: 2400MHz
17:58:05.833: Physical Cores: 10, Logical Cores: 10
17:58:05.833: Physical Memory: 16384MB Total
17:58:05.833: OS Name: macOS
17:58:05.833: OS Version: Version 12.1 (Build 21C52)
17:58:05.833: Rosetta translation used: true
17:58:05.833: Kernel Version: 21.2.0
17:58:05.842: hotkeys-cocoa: Getting keyboard keys failed
17:58:05.842: hotkeys-cocoa: Using layout 'com.apple.keylayout.Dvorak'
17:58:05.843: Current Date/Time: 2022-01-26, 17:58:05
17:58:05.843: Browser Hardware Acceleration: true
17:58:05.843: Qt Version: 5.15.2 (runtime), 5.15.2 (compiled)
17:58:05.843: Portable mode: false
17:58:05.965: OBS 27.2.0-beta4-23-g026987a2b (mac)
```

</details>

### Types of changes

- New feature (non-breaking change which adds functionality) 

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: libobs/obs-cocoa.m [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/sysctl.h>
+#include <sys/utsname.h>
```
Comment (gxalpha) [inline libobs/obs-cocoa.m:26]: Is this include requried?

File: libobs/util/platform-windows.c [hunk 2]
```diff
@@ -1088,9 +1088,24 @@ bool is_64_bit_windows(void)
 #endif
 }
 
+bool is_arm64_windows(void)
+{
+#if defined(_M_ARM64) || defined(_M_ARM64EC)
+	return true;
+#else
+	USHORT _, nativeMachine;
+	auto result = IsWow64Process2(GetCurrentProcess(), &_, &nativeMachine);
```
Comment (RytoEX) [inline libobs/util/platform-windows.c:1097]: The new C auto is not implemented until C23, which we do not currently support. This is currently the old C auto, which we do not recommend. Use the actual type here.

File: libobs/util/platform-windows.c [hunk 3]
```diff
@@ -1088,9 +1088,24 @@ bool is_64_bit_windows(void)
 #endif
 }
 
+bool is_arm64_windows(void)
+{
+#if defined(_M_ARM64) || defined(_M_ARM64EC)
+	return true;
+#else
+	USHORT _, nativeMachine;
```
Comment (RytoEX) [inline libobs/util/platform-windows.c:1096]: Please do not declare multiple items on a single line. Declare them separately.

Comment (RytoEX) [inline libobs/util/platform-windows.c:1096]: Please use a meaningful name. Please also do not start identifiers with an underscore, per the C standard section on reserved identifiers.

**Review by RytoEX — CHANGES_REQUESTED** [review]

**Pull request merged**